### PR TITLE
Add network attachment to gce instance

### PR DESCRIPTION
**Pull request by maxi-cit**
This PR enables support for network attachment at the gce instance network interfaces

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
google_compute_instance
```

### CODE REVIEW

**Comment by trodge** [comment]
Tests are failing with the following error:
```
2023-09-02T00:21:13.173Z [ERROR] sdk.proto: Response contains error diagnostic: tf_provider_addr=registry.terraform.io/hashicorp/google tf_proto_version=5.3 diagnostic_detail="" diagnostic_severity=ERROR diagnostic_summary="Error creating instance: googleapi: Error 400: Invalid value for field 'resource.networkInterfaces[1]': '{  \"networkAttachment\": \"https://www.googleapis.com/compute/beta/projects/ci-test-project-188019/reg...'. Instance must be in the same region as the network attachment., invalid" tf_rpc=ApplyResourceChange tf_resource_type=google_compute_instance tf_req_id=1bac79aa-4c87-e3ed-6d10-8018caf72a57
```

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb [hunk 1]
```diff
@@ -6829,3 +6909,188 @@ resource "google_compute_disk" "debian" {
...
+
+    subnetworks = [
+        google_compute_subnetwork.test-subnetwork2.self_link
+    ]
+}
+
+resource "google_compute_instance" "foobar" {
+	name         = "%{instance_name}"
+	machine_type = "e2-medium"
+	zone         = "us-east1-d"
```
Comment (trodge) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:6998]: I believe this needs to be a zone in us-central1.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb [hunk 2]
```diff
@@ -6829,3 +6909,188 @@ resource "google_compute_disk" "debian" {
...
+resource "google_compute_subnetwork" "test-subnetwork2" {
+	name          = "tf-test-compute-subnet2-%{suffix}"
+	ip_cidr_range = "10.3.0.0/16"
+	region        = "us-central1"
+	network       = google_compute_network.test-network.id
+}
+
+resource "google_compute_network_attachment" "test_network_attachment2" {
+	name = "tf-test-network-attachment2-%{suffix}"
+    region = "us-central1"
```
Comment (trodge) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:6986]: Please fix the indentation here.

**Review by trodge — CHANGES_REQUESTED** [review]

**Comment by maxi-cit** [comment]
Hello @trodge I updated the behavior of this field since this [API](https://cloud.google.com/compute/docs/reference/rest/beta/instances/updateNetworkInterface) is not able to update an interface's network attachment. This makes this field immutable and will require instance recreation
I also updated the related tests

**Review by trodge — APPROVED** [review]

**Pull request merged**